Netcracker/KubeMarine
# MANOPD-84415 MANOPD-86236 Plugin manifests enrichment by python

### Description
* It is necessary to rework enrichment of all supported plugins from jinja to yaml+python, the similar as it was done for `calico` in #279
* After the same PR many env variables stopped propagating to `calico-node` container.

### Solution
* Make env variables enrichment of `calico-node` container effectively the same as before #279 with the only exception:
    * Before the PR FELIX_IPV6SUPPORT was removed for IPv4 cluster. Now it is not removed and kept with `false` default value instead.
* Major refactoring of `calico` enrichment to re-use common approach in other plugins.
* Enrichment of `nginx-ingress-controller`, `kubernetes-dashboard`, and `local-path-provisioner` is reworked from jinja to python with the effectively same result. The two exceptions are listed below.
    * `nginx-ingress-controller` for k8s v1.21 - v1.24 now has `allow-snippet-annotations: "true"` in ConfigMap the same as for v1.25. The 'true' value is actually the [default](https://github.com/kubernetes/ingress-nginx/blob/controller-v1.2.0/docs/user-guide/nginx-configuration/configmap.md#allow-snippet-annotations) one.
    * `local-path-provisioner` had images and template related to different versions: 0.0.22/0.0.23 and 0.0.19 correspondingly. Now both image and original manifest is taken for 0.0.22/0.0.23 version. The following changes are included: https://github.com/rancher/local-path-provisioner/pull/177 and https://github.com/rancher/local-path-provisioner/pull/188
* Partially revert #228 as it is no longer necessary to hold default env variables in `defaults.yaml`.
* Old jinja templates are **not removed** for now as old inventories can have references to them. The removal is planned in future (release).

### How to apply
Run `migrate_kubemarine --force-apply calico_fixes`

### Test Cases

**TestCase 1**

Steps:

1. Configure calico env variables in inventory: CALICO_ROUTER_ID, IP_AUTODETECTION_METHOD, CALICO_IPV4POOL_CIDR, FELIX_LOGSEVERITYSCREEN, FELIX_USAGEREPORTINGENABLED
2. Run `install` procedure.

Results:

| Before | After |
| ------ | ------ |
| Env variables are absent in `calico-node` pods | The env variables are present |

**TestCase 2**

Steps:

1. Run `install` procedure in IPv6 cluster.

Results:

| Before | After |
| ------ | ------ |
| IPv6-related env variables are absent in `calico-node` pods | The env variables are present |

**TestCase 3**

Steps:

1. Install cluster using previous version of Kubemarine.
2. Check that some env variables are absent in `calico-node` pods (see TC1 and TC2)
3. Run `migrate_kubemarine --force-apply calico_fixes` using new version of Kubemarine.

Results:

| Before | After |
| ------ | ------ |
| Not applicable | The env variables are present |

### Regression testing

Steps:

1. Configure inventory:
    * Configure `calico`, `nginx-ingress-controller`, `kubernetes-dashboard`, and `local-path-provisioner` specifying different parameters.
    * Configure `pss` and `psp` admission implementations.
4. Run installation procedure.

ER: Kubemarine installs the plugins with the same final manifests as it was done by previous Kubemarine version. Final manifests can be seen in /etc/kubernetes folder.

### Checklist

#### Unit tests
Implemented unit tests for enrichment of all plugins.
